psydac/fem/vector.py

```diff
@@ -225,11 +224,11 @@ def pushforward_grid(self, grid, *fields, mapping=None, npts_per_cell=None):
                 ncells_i = len(self.spaces[0].breaks[i]) - 1
                 grid[i] = np.reshape(grid[i], newshape=(ncells_i, npts_per_cell[i]))
 
-            pushed_fields = self.pushforward_regular_tensor_grid(grid, *fields, mapping=mapping)
+            pushed_fields = self.pushforward_fields_regular_tensor_grid(grid, *fields, mapping=mapping)
             # return a list of list of C-contiguous arrays, one list for each field
             # with one array for each dimension.
-            return [[np.ascontiguousarray(pushed_fields[..., j, i]) for j in range(self._ldim)]
-                    for i in range(len(fields))]
+            return [[np.ascontiguousarray(pushed_fields[..., i, j]) for i in range(self._ldim)]
+                    for j in range(len(fields))]
```

Comment:
Can you please leave the indexing as before, and simply swap the two ranges? I think that's clearer

Reply:
Done d58dc62